### PR TITLE
added route to get rated cocktails for a user

### LINKED ISSUE #46: end point for all cocktails a user has rated
for the rated cocktails page

### PR DESCRIPTION
**Pull request by zachjamesgreen**
Added route get all user rated cocktails 

## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Rubocop has passed locally and any fixes were made for failures
- [x] CircleCI build is passing
- [x] All tests passing locally

## Pull request type
Please check the type of change your PR introduces:
- [x] Feature

## What is the current behavior?

Issue Number: 46
closes #46 

## What is the new behavior?

Added route get all user rated cocktails 

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by redferret — APPROVED** [review]

File: spec/requests/api/v1/cocktails/rated_cocktails_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      Rating.create!(user: @user1, cocktail: cocktail, stars: 1)
+    end
+
+    @cocktails2 = create_list(:cocktail, 3)
+    @cocktails2.each do |cocktail|
+      Rating.create!(user: @user2, cocktail: cocktail, stars: 1)
+    end
+  end
+
+  it 'returns a list of rated cocktails for the user' do
```
Comment (mkrumholz) [inline spec/requests/api/v1/cocktails/rated_cocktails_spec.rb:19]: Should we add some sad path testing here for if the user doesn't have rated cocktails to make sure it's returning what we expect?

File: app/models/user.rb [hunk 2]
```diff
@@ -3,4 +3,5 @@ class User < ApplicationRecord
   validates :email, uniqueness: true
 
   has_many :ratings, dependent: :destroy
+  has_many :cocktails, through: :ratings
```
Comment (mkrumholz) [inline app/models/user.rb:6]: Nice! This relationship makes the controller action really elegant.

Comment (markcyen) [inline app/models/user.rb:6]: nice addition

**Review by markcyen — APPROVED** [review]
Looks good.

File: spec/requests/api/v1/cocktails/rated_cocktails_spec.rb (hunk 1 above)
Comment (zachjamesgreen) [reply spec/requests/api/v1/cocktails/rated_cocktails_spec.rb:20]: added a test to make sure it sends an empty array in no cocktails are rated

**Pull request merged**